inc/domainrecordtype.class.php

```diff
@@ -42,46 +42,225 @@ class DomainRecordType extends CommonDropdown
+         'fields'    => [],
       ], [
          'id'        => 5,
          'name'      => 'MX',
-         'comment'   => 'Mail eXchange'
+         'comment'   => 'Mail eXchange',
+         'fields'    => [
+            [
+               'label'       => 'Priority',
+               'placeholder' => 'eg. 10',
+            ],
+            [
+               'label'       => 'Server',
+               'placeholder' => 'eg. mail.example.com',
+            ],
+         ],
       ], [
          'id'        => 6,
          'name'      => 'NS',
-         'comment'   => 'Name Server'
+         'comment'   => 'Name Server',
+         'fields'    => [],
       ], [
          'id'        => 7,
          'name'      => 'PTR',
-         'comment'   => 'Pointer'
+         'comment'   => 'Pointer',
+         'fields'    => [],
       ], [
          'id'        => 8,
          'name'      => 'SOA',
          'comment'   => 'Start Of Authority',
+         'fields'    => [
+            [
+               'label'       => 'Primary name server',
+               'placeholder' => 'eg. ns1.example.com',
+            ],
+            [
+               'label'       => 'Primary contact',
+               'placeholder' => 'eg. admin.example.com',
+            ],
+            [
+               'label'       => 'Serial',
+               'placeholder' => 'eg. 2020010101',
+            ],
+            [
+               'label'       => 'Zone refresh timer',
+               'placeholder' => 'eg. 86400',
+            ],
+            [
+               'label'       => 'Failed refresh retry timer',
+               'placeholder' => 'eg. 7200',
+            ],
+            [
+               'label'       => 'Zone expiry timer',
+               'placeholder' => 'eg. 1209600',
+            ],
+            [
+               'label'       => 'Minimum TTL',
+               'placeholder' => 'eg. 300',
+            ],
+         ],
       ], [
          'id'        => 9,
          'name'      => 'SRV',
-         'comment'   => 'Location of service'
+         'comment'   => 'Location of service',
+         'fields'    => [
+            [
+               'label'       => 'Priority',
+               'placeholder' => 'eg. 0',
+            ],
+            [
+               'label'       => 'Weight',
+               'placeholder' => 'eg. 10',
+            ],
+            [
+               'label'       => 'Port',
+               'placeholder' => 'eg. 5060',
+            ],
+            [
+               'label'       => 'Target',
+               'placeholder' => 'eg. sip.example.com',
+            ],
+         ],
       ], [
          'id'        => 10,
          'name'      => 'TXT',
-      'comment'    => 'Descriptive text'
+         'comment'   => 'Descriptive text',
+         'fields'    => [
+            [
+               'label'       => 'TXT record data',
+               'placeholder' => 'Your TXT record data',
+               'quote_value' => true,
```

Comment:
this one may be problematic, we need to check with powerdns plugin

Reply:
I checked, the plugin will handle that correctly.